dpbench/infrastructure/benchmark.py

```diff
@@ -54,9 +56,25 @@ def _setup_func(initialized_data, array_args, framework):
     return copied_args
 
 
+def _array_size(array: Any) -> int:
+    if isinstance(array, np.ndarray):
+        return array.size * array.itemsize
+    # for some reason runtime treat type as dpnp_array.dpnp_array
+    if isinstance(array, dpnp.dpnp_array.dpnp_array):
```

Comment:
I am not sure why we need to check for dpnp and dpctl.tensor types. For every framework we create input data as numpy ndarrays and copy the numpy data to dpnp and dpctl where ever it is necessary. We should be able to intercept at the level of numpy ndarray creation and get the array size.
The idea is to avoid adding domain information to infrastructure as much as possible.

Reply:
I totally agree with you. Just have a concert, that different test may have differ domain type. It may result in different sizes. Not sure if it would be the case, but a good thing to keep in mind.